### PR TITLE
Subscription mailer

### PR DESCRIPTION
**Pull request by jjgonecrypto**
Adding daily mailing functionality for subtrackr

### CODE REVIEW

File: app/models/subscription.rb [hunk 1]
```diff
@@ -18,13 +18,18 @@ class Subscription
...
   before_save :calculate_billing_and_notification_dates
 
-  def self.send_email
-      where(:frequency => "monthly", 
-	   :offset => ( Date.today + :notify_before).day).each do |subscription|
-	     #MemberMailer.subscription_notification(subscription.user, subscription).deliver
-	     puts subscription.service + ', ' + subscription.user.email
-	     subscription.save! #recalc the next bill
-	   end
+  def self.check_and_send_notifications
```
Comment (jonstorer) [inline app/models/subscription.rb:21]: this works, but is memory inefficient. so if/when it becomes a problem you're going to have to refactor

File: spec/models/subscription_spec.rb [hunk 2]
```diff
@@ -13,6 +13,11 @@
     it { should belong_to(:user).of_type(User) }
   end
 
+  after (:each) do
```
Comment (jonstorer) [inline spec/models/subscription_spec.rb]: if you don't wanna keep track of this.
https://github.com/jonstorer/safetext-api/blob/master/spec/spec_helper.rb

File: spec/models/subscription_spec.rb [hunk 3]
```diff
@@ -75,25 +80,24 @@
...
-       year = 2011
-       month = 12
-       day = 9 #this date is a friday - offset of 5
-       Timecop.freeze(Date.new(year,month,day)) do 
-	  subject.next_bill.should == Date.today + 6 
-          subject.notify_date.should == subject.next_bill - notify; 
-       end
+  context "ensure subscription mailer" do
+    before (:each) do
+       @bill = Date.today
```
Comment (jonstorer) [inline spec/models/subscription_spec.rb]: `before`s are usually intended for manipulating data that can't be "setup".

can you put these in `let(:variable)` ?

File: app/models/subscription.rb (hunk 1 above)
Comment (jjgonecrypto) [inline app/models/subscription.rb:21]: yea, noticed it wasn't perfect but couldn't come up with something off the top of my head... will keep it in mind

File: spec/spec_helper.rb [hunk 4]
```diff
@@ -33,3 +33,5 @@
   # rspec-rails.
   
 end
+
+require 'mocha'
```
Comment (jonstorer) [inline spec/spec_helper.rb]: this should be at the top. i didn't even know it would work at the bottom

Comment (jjgonecrypto) [inline spec/spec_helper.rb]: just wanted to ensure it would be included for all tests. will move it.

File: spec/models/subscription_spec.rb (hunk 3 above)
Comment (jjgonecrypto) [inline spec/models/subscription_spec.rb]: sure can.

File: spec/models/subscription_spec.rb (hunk 2 above)
Comment (jjgonecrypto) [inline spec/models/subscription_spec.rb]: ah right, didn't quite see the benefit of it before. =)

**Pull request merged**